network/endpoint_windows.go

```diff
@@ -518,6 +528,11 @@ func (nw *network) deleteEndpointImpl(_ netlink.NetlinkInterface, _ platform.Exe
 		return nil
 	}
 
+	if ep.HnsId == "" {
+		logger.Info("No HNS id found. Skip endpoint deletion", zap.Any("nicType", ep.NICType), zap.String("containerId", ep.ContainerID))
+		return nil
+	}
```

Comment:
when this can happen? Isn't this error scenario?

Reply:
deleteEndpointImpl()
if ep.HnsId == "" then return nil and log

Reply:
When `ep.HnsId` will be nil? isnt this error case.. lets remove this condition or throw error if hns id is empty